cartridges/plugins/wso2am-1.9.0/wso2am-startup-handler.py

```diff
@@ -0,0 +1,440 @@
+        port_mappings_str = values[self.CONST_PORT_MAPPINGS].replace("'", "")
+        app_id = values[self.CONST_APPLICATION_ID]
+        mb_ip = values[self.CONST_MB_IP]
+        service_name = values[self.CONST_SERVICE_NAME]
+        profile = os.environ.get(self.ENV_CONFIG_PARAM_PROFILE)
+        load_balancer_ip = os.environ.get(self.ENV_CONFIG_PARAM_LOADBALANCER_IP)
+        membership_scheme = values.get(self.ENV_CONFIG_PARAM_MEMBERSHIP_SCHEME)
+        clustering = values.get(self.ENV_CONFIG_PARAM_CLUSTERING, 'false')
+        my_cluster_id = values[self.CONST_CLUSTER_ID]
+        worker_hostname = os.environ.get(self.ENV_CONFIG_PARAM_WORKER_HOST_NAME)
+
+        # log above values
+        WSO2AMStartupHandler.log.info("Port Mappings: %s" % port_mappings_str)
+        WSO2AMStartupHandler.log.info("Application ID: %s" % app_id)
+        WSO2AMStartupHandler.log.info("MB IP: %s" % mb_ip)
+        WSO2AMStartupHandler.log.info("Service Name: %s" % service_name)
+        WSO2AMStartupHandler.log.info("Profile: %s" % profile)
+        WSO2AMStartupHandler.log.info("Membership Scheme: %s" % membership_scheme)
+        WSO2AMStartupHandler.log.info("Cluster ID: %s" % my_cluster_id)
+        WSO2AMStartupHandler.log.info("Clustering: %s" % clustering)
+        WSO2AMStartupHandler.log.info("Worker Hostname: %s" % worker_hostname)
+
+        # set sub-domain
+        sub_domain = None
+        if service_name.endswith(self.CONST_MANAGER):
+            sub_domain = self.CONST_MGT
+        elif service_name.endswith(self.CONST_WORKER):
+            sub_domain = self.CONST_WORKER
+        if sub_domain is not None:
+            self.export_env_var(self.ENV_CONFIG_PARAM_SUB_DOMAIN, sub_domain)
+
+        # export Proxy Ports as Env. variables - used in catalina-server.xml
+        self.set_proxy_ports(port_mappings_str)
+
+        # if CONFIG_PARAM_MEMBERSHIP_SCHEME is not set, set the private-paas membership scheme as default one
+        if membership_scheme is None:
+            membership_scheme = self.CONST_PPAAS_MEMBERSHIP_SCHEME
+            self.export_env_var(self.ENV_CONFIG_PARAM_MEMBERSHIP_SCHEME, membership_scheme)
+
+        if clustering == 'true' and membership_scheme == self.CONST_PPAAS_MEMBERSHIP_SCHEME:
+            # export Cluster_Ids as Env. variables - used in axis2.xml
+            service_list = None
+
+            if service_name in self.GATEWAY_SERVICES:
+                service_list = self.GATEWAY_SERVICES
+            elif service_name in self.PUB_STORE_SERVICES:
+                service_list = self.PUB_STORE_SERVICES
+
+            self.set_cluster_ids(app_id, service_name, my_cluster_id, service_list)
+            # export mb_ip as Env.variable - used in jndi.properties
+            if mb_ip is not None:
+                self.export_env_var(self.ENV_CONFIG_PARAM_MB_HOST, mb_ip)
+
+        if profile == self.CONST_KEYMANAGER:
```

Comment:
It's really good to see this implementation which uses the profile parameter. Great work! Shall we add few comments inside each block describing the configurations being done? Thanks